### PR TITLE
test for exception on missing host or socketPath 

### PR DESCRIPTION
**Pull request by johncoffee**

### CODE REVIEW

File: test/functional/connection/connection.ts [hunk 1]
```diff
@@ -81,6 +82,39 @@ describe("Connection", () => {
...
+                    dropSchemaOnConnection: false,
+                    schemaCreate: false,
+                    enabledDrivers: ["mysql"],
+                });
+            }
+            catch (e) {
+                didThrow = true;
+            }
+            expect(didThrow).to.be.true;
+        });
```
Comment (pleerock) [inline test/functional/connection/connection.ts]: btw you can do 

```
expect(() => {
     connection = getConnectionManager().create(<ConnectionOptions>{
                     driver: {
                         "type": "mysql",
                         "username": "test",
                         "password": "test",
                         "database": "test",
                     },
                     entities: [],
                     entitySchemas: [],
                     dropSchemaOnConnection: false,
                     schemaCreate: false,
                    enabledDrivers: ["mysql"],
                 });

}).to.throw(Error);

File: test/functional/connection/connection.ts [hunk 2]
```diff
@@ -81,6 +82,39 @@ describe("Connection", () => {
...
+    describe("establishing connection", function() {
+        let connection: Connection;
+        after(() => {
+            if (connection && connection.isConnected)
+                return connection.close();
+
+            return Promise.resolve();
+        });
+
+        it("should throw DriverOptionNotSetError when extra.socketPath is missing", function() {
```
Comment (pleerock) [inline test/functional/connection/connection.ts]: probably correct statement is `when both "extra.socketPath" and "host" are missing`

File: test/functional/connection/connection.ts [hunk 3]
```diff
@@ -81,6 +82,39 @@ describe("Connection", () => {
 
     });
 
+    describe("establishing connection", function() {
+        let connection: Connection;
+        after(() => {
+            if (connection && connection.isConnected)
+                return connection.close();
```
Comment (pleerock) [inline test/functional/connection/connection.ts]: will it even be connected in this test?

File: test/functional/connection/connection.ts (hunk 2 above)
Comment (johncoffee) [inline test/functional/connection/connection.ts]: Changed it.

File: test/functional/connection/connection.ts (hunk 3 above)
Comment (johncoffee) [inline test/functional/connection/connection.ts]: I left it to be overly secure. Now I removed it.

**Comment by pleerock** [comment]
Thank you!

**Pull request merged**